dotCMS/src/main/java/com/dotcms/cache/VanityUrlCacheImpl.java

```diff
@@ -0,0 +1,120 @@
+package com.dotcms.cache;
+
+import com.dotcms.util.VanityUrlUtil;
+import com.dotcms.vanityurl.model.CachedVanityUrl;
+import com.dotcms.vanityurl.model.VanityUrl;
+import com.dotmarketing.business.CacheLocator;
+import com.dotmarketing.business.DotCacheAdministrator;
+import com.dotmarketing.business.DotCacheException;
+import com.dotmarketing.util.Logger;
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * This class implements the cache for Vanity URLs.
+ * Is used to map the Vanity URLs path to the Vanity URL
+ * content
+ *
+ * @author oswaldogallango
+ */
+public class VanityUrlCacheImpl extends VanityUrlCache {
+
+    private static DotCacheAdministrator cache;
+
+    private static final String primaryGroup = "VanityURLCache";
+    private static final String cachedVanityUrlGroup = "cachedVanityUrlGroup";
+    // region's name for the cache
+    private static final String[] groupNames = {primaryGroup, cachedVanityUrlGroup};
+
+    public VanityUrlCacheImpl() {
+        cache = CacheLocator.getCacheAdministrator();
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this assignment of "cache". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3010)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Synchronize this lazy initialization of 'cache' [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2444)